File: LeetCode/53/main.py
class Solution:
    def maxSubArray(self, arr: List[int]) -> int:
        curSum = 0
        maxSum = float("-inf")
        
        for i in range(len(arr)) :
            curSum = max(arr[i], curSum + arr[i])
            maxSum = max(maxSum, curSum)
        
        return maxSum    

File: LeetCode/53/main.cpp
class Solution {
public:
  int maxSubArray(vector<int> &arr) {
    int n = arr.size(), sum = 0, ans = INT_MIN;
    for (int i = 0; i < n; i++) {
      sum += arr[i];
      if (ans < sum) {
        ans = sum;
      }
      if (sum < 0) {
        sum = 0;
      }
    }
    return ans;
  }
};